crates/wasmtime/src/config.rs

```diff
@@ -1592,18 +1597,25 @@ impl Config {
                 .insert("enable_probestack".into());
         }
 
-        if self.native_unwind_info ||
-             // Windows always needs unwind info, since it is part of the ABI.
-             target.operating_system == target_lexicon::OperatingSystem::Windows
+        if self.profiling_strategy != ProfilingStrategy::None
+            || target.operating_system == target_lexicon::OperatingSystem::Windows
         {
-            if !self
-                .compiler_config
-                .ensure_setting_unset_or_given("unwind_info", "true")
-            {
-                bail!("compiler option 'unwind_info' must be enabled profiling");
+            unsafe {
+                self.cranelift_flag_set("unwind_info", "true");
             }
         }
 
+        const DEFAULT_UNWIND_INFO: bool = true;
+        if !self.compiler_config.ensure_setting_unset_or_given(
+            "unwind_info",
+            &self
+                .native_unwind_info
+                .unwrap_or(DEFAULT_UNWIND_INFO)
+                .to_string(),
+        ) {
+            bail!("`native_unwind_info` cannot be disabled when profiling or on Windows");
+        }
```

Comment:
Given the new information that profiling doesn't affect this setting at all, I think the structure should be more like:

```rust
if let Some(unwind_requested) = self.native_unwind_info {
    if !self.compiler_config.ensure_setting_unset_or_given("unwind_info", unwind_requested.to_string()) {
        bail!("incompatible settings requested for Cranelift and Wasmtime unwind-info settings");
    }
}

if target.operating_system == target_lexicon::OperatingSystem::Windows {
    if !self.compiler_config.ensure_setting_unset_or_given("unwind_info", "true") {
        bail!("`native_unwind_info` cannot be disabled on Windows");
    }
}
```

Reply:
Updated!